src/cache.jl

```diff
@@ -173,7 +206,18 @@ end
     job = CompilerJob(src, cfg)
 
     asm = nothing
-    # TODO: consider loading the assembly from an on-disk cache here
+    # can we load from the disk cache?
+    if disk_cache()
```

Comment:
Can we use `@static` here?

Reply:
I was frustrated by the need to recompile GPUCompiler to turn caching on and off.

I originally had it be a compile time preference which is what we would need for it to be `@static`

Reply:
I didn't realize we had non compile-time preferences...